src/common/SearchBar.js

```diff
@@ -0,0 +1,66 @@
+import Head from "next/head";
+import Image from "next/image";
+import { Inter } from "next/font/google";
+// import styles from "@/styles/Home.module.css";
+import FooterBar from "@/common/FooterBar";
+import ReusableNavLink from "@/common/ReusableNavLink";
+import Link from 'next/link';
+import Data from "../mock-data.json"
+import styles from "@/styles/SearchBar.module.css";
+import {useState} from 'react';
+
+
+function SearchBar({ placeholder, data }) {
+
+    const [filteredData, setFilteredData] = useState([]);
+
+    const handleFilter = (event) => {
+        const searchWord = event.target.value
+        const newFilter = data.filter((value) => {
+            return value.name.toLowerCase().includes(searchWord.toLowerCase());
+        });
+        if (searchWord === "") {
```

Comment:
Not absolutely necessary, but you could consider a ternary expression instead of this `if-else` block:

```suggestion
searchWord === "" ? setFilteredData([]) : setFilteredData(newFilter);
```